pkg/kubelet/metrics/metrics.go

```diff
@@ -917,6 +922,40 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+
+	// DRAOperationsDuration tracks the duration of the DRA PrepareResources and UnprepareResources requests.
+	DRAOperationsDuration = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           DRAOperationsDurationKey,
+			Help:           "Duration in seconds of the DRA operations (PrepareResources and UnprepareResources).",
+			Buckets:        metrics.DefBuckets,
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"operation_name"},
```

Comment:
I suggest to add boolean isError as a dimension.

Reply:
So far separate errors_total metric was the only approach used for Kubelet, e.g.
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L80
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L95
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L99
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L101
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L105
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L112
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L116
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L120

I just followed established practice. Is DRA so special that we need to do it differently?

Reply:
I don't think that @SergeyKanzhelev was suggesting that DRA is special. He just pointed out that the metric could be simplified. Whether that considered the usability aspect (https://github.com/kubernetes/kubernetes/pull/127146#discussion_r1807676380) or consistency (above) I don't know.

Adding a new metric is an opportunity to re-evaluate previous design choices. There's enough bad code also in Kubernetes that "I'm doing the same as everyone else" raises questions. I myself don't have a strong opinion here and will defer to people responsible for kubelet (@SergeyKanzhelev) or SIG Instrumentation (@logicalhan) for guidance.

Reply:
There are a lot of bad code in Kubernetes for other reasons as well. "Let's do it differently" could be one of those. I tend to think that existing metrics are used by a lot of people and they'd probably tell us if they're not happy about how they're done.
Anyway, let's may be give @SergeyKanzhelev an opportunity to speak for himself?

Reply:
I understand it might not be what we use in other metrics. But I would argue that what I suggest is materially better as it allows to better analyze metrics. Also it is aligned more with how OpenTelemetry generally defines the RPC semantic: https://opentelemetry.io/docs/specs/semconv/rpc/grpc/ (note, error code duration on the duration metrics and the duration metric being the ONLT thing needed - all other things like errors and total can be derived from it.

Reply:
I believe this PR introduces the same semantic for `grpc` [metrics](https://github.com/kubernetes/kubernetes/pull/127146/files#diff-7a99719139dd480a780b106a988e3175c2a5e72d48cc8399666843a8e631221dR958). However, this metric(`DRAOperationsDuration`) is more about Kubelet operations end errors. That's why I followed Kubelet pattern for it.

Let me know if you still think it makes sense to add `isError` dimension.

Reply:
done